### ISSUE TITLE
Class  does not exist in 3.2

### ISSUE THREAD

**Issue by mrardon** [schmittjoh/JMSDiExtraBundle#262]
I tried to run my application with updated Symfony 3.2.0-beta 1 and a reflection error is thrown with message "Class  does not exist". It seems to fail here: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/ContainerBuilder.php#L887

It seems to fail on a service dealing with ID "cache.annotations".

Steps to reproduce with Symfony Standard 3.2.0-beta1:

* `symfony new test 3.2.0-beta1`
* Ensure standard edition works by running a command `bin/console debug:container`
* Install/require latest di-extra-bundle `composer require jms/di-extra-bundle` (it chose 1.8.1)
* Add to AppKernel.php after `new AppBundle\AppBundle(),`:
```
            new JMS\DiExtraBundle\JMSDiExtraBundle($this),
            new JMS\AopBundle\JMSAopBundle(),
```
* Add configuration to config.yml:
```
jms_di_extra:
    locations:
        bundles:
            - AppBundle
```
* Create  src/AppBundle/Service/TestService.php
```
<?php

namespace AppBundle\Service;

use JMS\DiExtraBundle\Annotation as DI;

/**
 * @DI\Service("test_service")
 */
class TestService
{

}
```
* Run a command `bin/console debug:container` fails with `Class  does not exist` because class appears to be set to NULL

Any ideas?

Edit: add missing step about creating TestService.php

**Comment by mrardon** [comment]
Here is an exception trace:
```
() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:887
ReflectionClass->__construct() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:887
Symfony\Component\DependencyInjection\ContainerBuilder->createService() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:443
Symfony\Component\DependencyInjection\ContainerBuilder->get() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:946
Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:943
Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:868
Symfony\Component\DependencyInjection\ContainerBuilder->createService() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:443
Symfony\Component\DependencyInjection\ContainerBuilder->get() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:946
Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:943
Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:868
Symfony\Component\DependencyInjection\ContainerBuilder->createService() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:443
Symfony\Component\DependencyInjection\ContainerBuilder->get() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:946
Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:943
Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:868
Symfony\Component\DependencyInjection\ContainerBuilder->createService() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:443
Symfony\Component\DependencyInjection\ContainerBuilder->get() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:427
Symfony\Component\DependencyInjection\ContainerBuilder->get() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:946
Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:943
Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:868
Symfony\Component\DependencyInjection\ContainerBuilder->createService() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:443
Symfony\Component\DependencyInjection\ContainerBuilder->get() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:427
Symfony\Component\DependencyInjection\ContainerBuilder->get() at /test/vendor/jms/metadata/src/Metadata/Driver/LazyLoadingDriver.php:23
Metadata\Driver\LazyLoadingDriver->loadMetadataForClass() at /test/vendor/jms/metadata/src/Metadata/MetadataFactory.php:103
Metadata\MetadataFactory->getMetadataForClass() at /test/vendor/jms/di-extra-bundle/DependencyInjection/Compiler/AnnotationConfigurationPass.php:99
JMS\DiExtraBundle\DependencyInjection\Compiler\AnnotationConfigurationPass->handlePattern() at /test/vendor/jms/di-extra-bundle/DependencyInjection/Compiler/AnnotationConfigurationPass.php:62
JMS\DiExtraBundle\DependencyInjection\Compiler\AnnotationConfigurationPass->process() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/Compiler.php:114
Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at /test/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:565
Symfony\Component\DependencyInjection\ContainerBuilder->compile() at /test/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:484
Symfony\Component\HttpKernel\Kernel->initializeContainer() at /test/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:116
Symfony\Component\HttpKernel\Kernel->boot() at /test/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:68
Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /test/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:122
Symfony\Component\Console\Application->run() at /test/bin/console:27
```

**Comment by mrardon** [comment]
Looks like there is a PR that may help in symfony core. A similar issue was found with VichUploadBundle. I am unable to do a test to check if this works now but wanted to point it out incase someone was so inclined.

https://github.com/symfony/symfony/pull/20537

**Comment by mrardon** [comment]
Looks like that is the cause, checked with a patch from the above pull request. Going to close this and hope they approve the pull request. 🙏

**Comment by mrardon** [comment]
Looks like the fix for this broke some other things (tags) of parents, any idea on possible work arounds? Will have to stop using this bundle to upgrade to 3.2. Any ideas where to look to fix the issue, I'm not sure if I understand 100% what is causing this even with the fix presented.

**Comment by nicolas-grekas** [comment]
Fixed for now in 3.2 with https://github.com/symfony/symfony/pull/20601

**Comment by mrardon** [comment]
Just tested out with 3.2.0-RC2 and all seems well, thanks @nicolas-grekas for fix for now.
